clang/test/CodeGenSYCL/matrix.cpp

```diff
@@ -0,0 +1,31 @@
+// RUN: %clang_cc1 -triple spir64-unknown-unknown -disable-llvm-passes -emit-llvm %s -o - -no-opaque-pointers | FileCheck %s
+#include <stddef.h>
+#include <stdint.h>
+
+namespace __spv {
+  template <typename T, size_t R, size_t C, uint32_t U, uint32_t S>
+  struct __spirv_JointMatrixINTEL;
+}
+
+// CHECK: @_Z2f1{{.*}}(%spirv.JointMatrixINTEL._float_5_10_0_1
+void f1(__spv::__spirv_JointMatrixINTEL<float, 5, 10, 0, 1> *matrix) {}
+
+// CHECK: @_Z2f2{{.*}}(%spirv.JointMatrixINTEL._long_10_2_0_0
+void f2(__spv::__spirv_JointMatrixINTEL<uint64_t, 10, 2, 0, 0> *matrix) {}
```

Comment:
Can you add a test case for the larger that 64 int sizes too?

Reply:
`__int128` is not legal on SPIR-V backend, and I'm not sure they are legal types for the JointMatrixINTEL type in any case.

Reply:
It's legal to use `ExtInt` in SPIR-V with an arbitrary width under `ArbitraryPrecisionIntegersINTEL` capability. Also there are no restrictions about integer width for element type of the JointMatrix.
So may be it worth it add a case, like:
`void f2(__spv::__spirv_JointMatrixINTEL<_ExtInt(42), 10, 2, 0, 0> *matrix) {}`
to see if we get `._i42_10_2_0_0` postfix.

Reply:
I did find that it was possible to use `_ExtInt` in a test and added it in a later patch, albeit for `_ExtInt(128)`.